### DISCUSSION TITLE
Error found "Cannot use import statement outside a module"

### DISCUSSION THREAD

**Discussion by RaulFerreiraSilva** [typesnippet/antd-phone-input#104] [Q&A]
I congratulate you for the work developed, a very useful tool.
but I hereby inform you that when I update my page or try to build I come across this error "Cannot use import statement outside a module"

![imagem_2024-08-07_170854829](https://github.com/user-attachments/assets/6db72764-a5c8-4486-821a-09dbe959e8e0)

I noticed that this error also occurs in the testing part of the tool:

![imagem_2024-08-07_171021152](https://github.com/user-attachments/assets/7b5033e1-db16-4ea3-a676-72f23bad91ca)

**Comment by CaioCintra** [comment]
Same issue here

**Comment by ArtyomVancyan** [comment]
@RaulFerreiraSilva, I am not sure about the way RunKit imports or uses the package in its isolated environment, and that's why I need a code to reproduce the issue and see what the problem is. Could you please provide me with some code where the problem occurs?

> I come across this error "Cannot use import statement outside a module"

Usually, this error appears by misconfigured `tsconfig.json`, so you can also run one of the [examples](https://github.com/typesnippet/antd-phone-input/tree/master/examples) locally and check if it works with your `tsconfig.json` configuration. Compare the differences and find the cause.

**Reply by berezh** [reply]
I'm working with `nextjs` and resolved this issue with dynamic load:

`const PhoneInput = dynamic(() => import("antd-phone-input"), {
  ssr: false,
});`

**Reply by ArtyomVancyan** [reply]
Hi @berezh, can you please provide a sample application so I can reproduce the issue and fix it?

**Reply by berezh** [reply]
React project with below dependencies:
"next": "^13.1.1"
"react": "18.2.0",

Node version - 18

**Reply by ArtyomVancyan** [reply]
@berezh, I have created a NextJS application [sample](https://github.com/ArtyomVancyan/antd-phone-input-fork/tree/master/examples/antd5.x-next) to test the package. I faced an issue on `yarn build` with the provided dependency versions. It was enough to upgrade the `react` and `react-dom` versions (in `18.2.0`, there was a problem with types), and the problem was fixed. Please try to upgrade your dependency versions to `^18.3.1` and verify that it works for you.

**Reply by berezh** [reply]
upgraded "react": "^18.3.1". issue still exists. I think on your example you should use `InputPhone` inside antd form to reproduce the issue

**Reply by ArtyomVancyan** [reply]
I tried it the following way, and it still works for me. Please fork the sample I have shared with you and perform the necessary changes so I can obtain the error.

```jsx
import dynamic from "next/dynamic";

const PhoneInput = dynamic(() => import("antd-phone-input"), {ssr: false});
const FormItem = dynamic(() => import("antd/es/form/FormItem"), {ssr: false});
const Form = dynamic(() => import("antd/es/form"), {ssr: false});

export default function Home() {
    return (
        <Form>
            <FormItem name="phone">
                <PhoneInput/>
            </FormItem>
        </Form>
    )
}
```
